skrub/_reporting/_table_report.py

```diff
@@ -197,6 +200,43 @@ def _repr_mimebundle_(self, include=None, exclude=None):
     def _repr_html_(self):
         return self._repr_mimebundle_()["text/html"]
 
+    def write_html(self, file):
+        """Store the report into an HTML file.
+
+        Parameters
+        ----------
+        file : str, pathlib.Path or file object
+            The file object or path of the file to store the HTML output.
+        """
+        html = self.html()
+        if isinstance(file, (str, Path)):
+            with open(file, "w", encoding="utf8") as stream:
+                stream.write(html)
+            return
+        try:
+            file.write(html.encode("utf-8"))
+            return
+        except TypeError:
+            pass
+
+        print(getattr(file, "encoding", None))
+        if (encoding := getattr(file, "encoding", None)) is not None:
+            try:
+                assert codecs.lookup(encoding).name == "utf-8"
+            except (AssertionError, LookupError):
+                raise ValueError(
+                    "If `file` is a text file it should use utf-8 encoding; got:"
+                    f" {encoding!r}"
+                )
+        elif locale.getencoding().lower() != "utf-8":
+            # when encoding=None, it will default on the platform-specific encoding
+            # raise if not utf-8
+            raise ValueError(
+                f"Platform encoding is not utf-8; got {locale.getencoding()}"
+            )
```

Comment:
```suggestion
```

I think we can remove this block. if the encoding we find is `None` that does not indicate that there is an error or that the locale default encoding is being used: for example `report.write_html(io.StringIO()))` will work as expected regardless of the locale. Moreover if the user used `open` without providing an explicit encoding, using the locale's default, we would be in the previous case: the returned file object will have its encoding set explicitly to the default.

removing it will also remove the code coverage warning